web/base_viewer.js

```diff
@@ -63,8 +78,13 @@ function PDFPageViewBuffer(size) {
       data.shift().destroy();
     }
   };
-  this.resize = function(newSize) {
+  this.resize = function(newSize, setToKeep) {
```

Comment:
I can add clarifying comments, but to answer here: for current uses of `resize`, I don't think it's possible for `setToKeep.size` to exceed `newSize` (since `newSize` is never less than the number of visible pages, and `setToKeep` should be precisely the visible pages), but if it somehow is possible, discarding some of those pages is I guess the right thing? The buffer size after `resize` should definitely be `newSize`; that's the invariant. `setToKeep` is meant to be advisory (even though, as I said, I don't expect it to ever be in conflict with `newSize`).